### PR TITLE
ci: add pr title format validation

### PR DESCRIPTION
**Pull request by voxpelli**
**What is the purpose of this pull request?**

To assist #508 / #506 by ensuring that PR-title's are always valid conventional commit format before merging

**What changes did you make? (Give an overview)**

Added a pinned [mtfoley/pr-compliance-action](https://github.com/mtfoley/pr-compliance-action) similar to what I use at eg. neostandard: https://github.com/neostandard/neostandard/blob/main/.github/workflows/compliance.yml

### CODE REVIEW

**Comment by scagood** [comment]
This may be better as a branch ruleset :thinking:

Then again, I dont know if this org is an 'Enterprise' org on github?

(This should be the link) https://github.com/eslint-community/eslint-plugin-promise/settings/rules/new?target=branch

Eg here is one in one of my repos:
image

File: .github/workflows/compliance.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    types: [opened, edited, reopened]
+
+permissions:
+  pull-requests: write
+
+jobs:
+  compliance:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: mtfoley/pr-compliance-action@11b664f0fcf2c4ce954f05ccfcaab6e52b529f86
```
Comment (scagood) [inline .github/workflows/compliance.yml:14]: I think it'd be good to disable `issue-labels-auto-close`, and add the same exceptions that you added in neostandard for `renovate`

**Comment by voxpelli** [comment]
@scagood I believe that's only enforced in enterprise organizations

File: .github/workflows/compliance.yml (hunk 1 above)
Comment (voxpelli) [reply .github/workflows/compliance.yml:14]: 👍 This report uses dependabot currently though and I can't remember why/when the ignores are handy 😅

Comment (voxpelli) [reply .github/workflows/compliance.yml:14]: @scagood `issue-labels-auto-close` will not have to be disabled as no `issue-labels` has been set :)

**Review by scagood — APPROVED** [review]
LGTM :)

**Comment by voxpelli** [comment]
I added it to the ruleset now, but yeah, not enforced :)

Skärmavbild 2024-07-24 kl  16 24 08

**Pull request merged**